digdag-tests/src/test/java/acceptance/RequireIT.java

```diff
@@ -425,4 +449,85 @@ private void prepareForChildWF(Path childOutFile)
             Files.write(projectDir.resolve("child.dig"), child.getBytes("UTF-8"));
         }
     }
+
+    @Test
+    public void testRequireHasTargetSessionIdEvenWhenJustWaiting()
```

Comment:
I guess the purpose of this test is that we can get correct session_id and attempt_id when the session of the child already exists and require> will not kick by itself. So change the test name may be better.

In addition, this implementation does not assure parent will wait for child or not. Because the child workflow is very short and it is not certain whether the status of child is running or stopped when the parent session starts.
So 'waiting' is not correct in my understand.

Reply:
> I guess the purpose of this test is that we can get correct session_id and attempt_id when the session of the child already exists and require> will not kick by itself

Yes 👍 

> 'waiting' is not correct in my understand

I misunderstood the behavior, sorry. The parent **doesn't wait** for the child unless the parent actually kicks the child.
I will rename the test name and delete assertions that are based on the assumption the child finished successfully.

Reply:
Sorry, I misread your comment... 🙈 
The parent always waits for the child. It's independent of whether the parent kicks the child or not.
Will just rename the test case.

Reply:
done: https://github.com/hkdnet/digdag/commit/1a136ba7c377fb64bbfd57333cda7216aa287894